src/validateRegisterForm.test.js

```
@@ -26,5 +26,119 @@ describe(`Function 'validateRegisterForm':`, () => {
+  });
+
+  it(`should return error for valid email and password of
+   length 10 without digit`, () => {
+    const invalidPassword = validateRegisterForm('test@mail.com', 'P@ssword!');
+
+    expect(invalidPassword.code).toBe(422);
+    expect(invalidPassword.message).toBe('Password is invalid.');
+  });
+
+  it(`should return error for valid email and password greater
+   than 16 characters`, () => {
+    const invalidPassword = validateRegisterForm('test@mail.com',
+      'P@ssword123456789!');
+
+    expect(invalidPassword.code).toBe(422);
+    expect(invalidPassword.message).toBe('Password is invalid.');
+  });
+
+  it(`should return error for invalid email without '@' symbol`, () => {
+    const invalidEmail = validateRegisterForm('testmail.com', 'P@ssword1!');
+
+    expect(invalidEmail.code).toBe(422);
+    expect(invalidEmail.message).toBe('Email is invalid.');
+  });
+
+  it(`should return error for invalid email starting with '.'`, () => {
+    const invalidEmail = validateRegisterForm('.test@mail.com', 'P@ssword1!');
+
+    expect(invalidEmail.code).toBe(422);
+    expect(invalidEmail.message).toBe('Email is invalid.');
+  });
+
+  it(`should return error for invalid email ending with '.'`, () => {
+    const invalidEmail = validateRegisterForm('test@mail.com.', 'P@ssword1!');
+
+    expect(invalidEmail.code).toBe(200);
+    expect(invalidEmail.message).toBe('Email and password are valid.');
+  });
+
+  it(`should return error for invalid email with consecutive dots`,
+    () => {
+      const invalidEmail = validateRegisterForm('test..mail@mail.com',
+        'P@ssword1!');
+
+      expect(invalidEmail.code).toBe(422);
+      expect(invalidEmail.message).toBe('Email is invalid.');
+    });
+
+  it(`should return error for invalid email with top-level
+   domain starting with a dot`, () => {
+    const invalidEmail = validateRegisterForm('test@mail.com.', 'P@ssword1!');
+
+    expect(invalidEmail.code).toBe(200);
```

The test description says it should return an error, but the test expects a success code (200). Update the test description or the expected code accordingly.